modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1515,17 +1515,31 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
   /**
    * Get the current market price from a third party to be used for recovery
    */
-  getRecoveryMarketPrice(): Bluebird<string> {
+   getRecoveryMarketPrice(): Bluebird<string> {
     const self = this;
     return co<string>(function *getRecoveryMarketPrice() {
-      const bitcoinAverageUrl = config.bitcoinAverageBaseUrl + self.getFamily().toUpperCase() + 'USD';
-      const response = yield request.get(bitcoinAverageUrl).retry(2).result();
-
-      if (response === null || typeof response.last !== 'number') {
-        throw new Error('unable to reach BitcoinAverage for price data');
+      // TODO This map should be moved to the place where the list of families is defined
+      const familyNamesToCoinGeckoIds = new Map()
+        .set('BTC', 'bitcoin')
+        .set('LTC', 'litecoin')
+        .set('BCH', 'bitcoin-cash')
+        .set('ZEC', 'zcash')
+        .set('DASH', 'dash');
+      const coinGeckoId = familyNamesToCoinGeckoIds.get(self.getFamily().toUpperCase());
+      const coinGeckoUrl = config.coinGeckoBaseUrl + `simple/price?ids=${coinGeckoId}&vs_currencies=USD`;
```

Comment:
What happens if we don't find an entry in the map for the coin family?

Reply:
I added a check for that + throwing an error if a family is unknown.